Update Goreleaser

m-timmermann opened the pull request:
This PR updates the goreleaser.
It also removes deprecated keys

m-timmermann commented on .goreleaser.yml:
```diff
@@ -27,12 +27,6 @@ checksum:
   name_template: '{{ .ProjectName }}_checksums.txt'
 archives:
 - name_template: '{{ .ProjectName }}_{{ .Os }}_{{ .Arch }}{{ if .Arm }}v{{ .Arm }}{{ end }}'
-  replacements:
-    darwin: Darwin
-    linux: Linux
-    windows: Windows
-    386: i386
-    amd64: x86_64
```
Not sure about this part.
**I do not know if this part is needed or not.** 
see https://goreleaser.com/deprecations/#archivesreplacements for how it can be reimplemented

rwenz3l replied:
I think this is fine. The section was never really necessary in the first place, since the values listed here were actually part of the default anyway. The new default will also cover what we need:

https://github.com/goreleaser/goreleaser/blob/2498ea70290792c2537d98dbbdde0dd5ec07c2d6/internal/pipe/archive/archive.go#L75-L78

which yields:

```
{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}{{ with .Arm }}v{{ . }}{{ end }}{{ with .Mips }}_{{ . }}{{ end }}{{ if not (eq .Amd64 "v1") }}{{ .Amd64 }}{{ end }}
```

vs current:

```
'{{ .ProjectName }}_{{ .Os }}_{{ .Arch }}{{ if .Arm }}v{{ .Arm }}{{ end }}'
```

I would say keep it as is or just remove the template and use the default.

rwenz3l commented on .github/workflows/release.yml:
Why is the yaml so off in this? Usually you do:

```yaml
- name: "Checkout"
  uses: actions/checkout@v4
```

but here it is:

```yaml
-
  name: Checkout
  uses: actions/checkout@v2
```

also now that I see it: the version can be bumped on the actions.

m-timmermann commented:
Fixed the formatting.
Updated the version for go, actions/checkout@v4 and actions/setup-go@v5

The moment i update goreleaser/goreleaser-action@v1 the workflow wants me to add a tag or specifie ---snpashot in the ags for the goreleaser 
`⨯ release failed after 0s                  error=git doesn't contain any tags. Either add a tag or use --snapshot`

see https://github.com/meplato/store2-go-client/actions/runs/8451327839/job/23149434753#step:4:23

rwenz3l commented:
@m-timmermann the fetch-depth of the checkout action is 1 by default, so unless your latest commit contains a tag, it will rightfully complain. Usually you trigger this workflow on tags only, and in that case it should work. If you want to ensure there is a tag to build upon, use fetch-depth=0.

rwenz3l commented on .github/workflows/release.yml:
```diff
@@ -8,23 +8,16 @@ jobs:
-      -
-        name: Set up Go
-        uses: actions/setup-go@v1
+      - name: Checkout
+        uses: actions/checkout@v4
```
maybe add fetch-depth: 0 if you are unsure about tags being present. In the default case (when you release on tags, there will be a tag associated to the ref being checked out by git. You can test this by creating a dummy-tag on your latest commit in this PR.

XoseRamon commented on .github/workflows/release.yml:
```diff
@@ -8,23 +8,16 @@ jobs:
-        name: Run GoReleaser
+          go-version: 1.21
+      - name: Run GoReleaser
         uses: goreleaser/goreleaser-action@v1
```
In think go releaser is at version 5 or something like that already

XoseRamon commented on .github/workflows/release.yml:
```diff
@@ -8,23 +8,16 @@ jobs:
-          go-version: 1.18.x
-      -
-        name: Run GoReleaser
+          go-version: 1.21
```
You can go with 1.22, since we even have a .1 release

XoseRamon reviewed (commented):
check the version of go-releaser. Otherwise, this LGTM

m-timmermann commented:
Updated goreleaser version
Updated go version

The release workflow will now only be triggered if a tag was pushed.
